experiment/measurer.py

```diff
@@ -89,15 +97,30 @@ def measure_loop(experiment: str, max_total_time: int):
         # Using Multiprocessing.Queue will fail with a complaint about
         # inheriting queue.
         q = manager.Queue()  # pytype: disable=attribute-error
+
+        segment_list = manager.list(  # pytype: disable=attribute-error
+            [experiment_specific_df_container.segment_df])
+
+        function_list = manager.list(  # pytype: disable=attribute-error
+            [experiment_specific_df_container.function_df])
+
         while True:
             try:
                 # Get whether all trials have ended before we measure to prevent
                 # races.
                 all_trials_ended = scheduler.all_trials_ended(experiment)
 
-                if not measure_all_trials(experiment, max_total_time, pool, q):
+                if not measure_all_trials(experiment, max_total_time, pool, q,
+                                          segment_list, function_list):
                     # We didn't measure any trials.
                     if all_trials_ended:
+                        # Concatenate all data frames in the segment and
+                        # function multiprocessing list.
+                        experiment_specific_df_container.segment_df = (
```

Comment:
The trade-off here is that, this might be bit intensive in terms of memory, as the size data frame grows over time and is larger for longer experiments.

Printing ```segment_df.memory_usage(index=True, deep=True).sum()``` on the logger is in one way you could keep track of overall memory consumption.

Reply:
@BharathMonash Can you check the memory consumption for our local setup? The dispatcher runs on an n1-highmem-96 GCP instance, which has 624GB of main memory. If memory is a problem, we can merge and prune after each iteration of the measurer loop when the async call returns. Or we can write trial-specific CSVs to disk throughout, and load, merge and prune at the end.

Reply:
The memory consumption for the experiment config mentioned above in the PR description with the way that we are currently generating data is about 3.84 mb. With same, for a 24 hour experiment with 20 benchmarks,15 fuzzers and 10 trials the estimated memory consumption is ~ 8.23 GB. Pruning after each measure cycle reduces the estimate on a full experiment to 7.99 GB.